src/server/api/routers/posts.ts

```diff
@@ -75,6 +75,25 @@ export const postRouter = createTRPCRouter({
     return post;
   }),
 
+  delete: privateProcedure
+    .input(z.object({ id: z.string() }))
+    .mutation(async ({ ctx, input}) => {
+      const postId = input.id
+
+      const post = await ctx.db.post.findUnique({ where: {id: postId}})
```

Comment:
```suggestion
const post = await ctx.db.post.findUniqueOrThrow({ where: {id: postId}})
      .catch(()=> {
        throw new TRPCError({
          code: "NOT_FOUND",
          message: "There is no post"
        })
      })
```

`findUniqueOrThrow`を使うことでPostがnon nullableになります！
エラーメッセージをカスタマイズするには、`catch`を使用してあげることで86行目のような書き方ができます！

[Prisma Reference](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#findfirstorthrow)